monitoring/bin/create_elasticsearch_datasource.sh

```diff
@@ -0,0 +1,203 @@
+
+# Convert namespace and tenant to all lower-case
+tenantNS=$(echo "$tenantNS"| tr '[:upper:]' '[:lower:]')
+tenant=$(echo "$tenant"| tr '[:upper:]' '[:lower:]')
+
+# Check for parameters - set with cluster or tenant
+if [ -z "$tenantNS" ] && [ -z "$tenant" ]; then
+   cluster="true"
+elif [ -n "$tenant" ]; then
+    nst="${tenantNS}_${tenant}"
+else
+   nst="$tenantNS"
+fi
+
+# Check to see if monitoring/Viya namespace provided exists and components have already been deployed
+if [ "$cluster" == "true" ]; then
+    log_info "Checking for Grafana pods in the $MON_NS namespace ..."
+    if [[ $(kubectl get pods -n $MON_NS -l app.kubernetes.io/instance=v4m-prometheus-operator -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+        log_error "No monitoring components found in the [$MON_NS] namespace."
+        log_error "Monitoring needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+        exit 1
+    else
+        log_debug "Monitoring found in $MON_NS namespace.  Continuing."
+    fi
+else
+    log_info "Checking the [$tenantNS] namespace for monitoring deployment for the [$tenant] tenant ..."
+    if [[ $(kubectl get pods -n $tenantNS -l app.kubernetes.io/instance=v4m-grafana-$tenant -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+        log_error "No monitoring components were found for $tenant in the $tenantNS namespace."
+        log_error "Monitoring needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+        exit 1
+    else
+        log_debug "Monitoring deployment was found for the $tenant tenant in the $tenantNS namespace.  Continuing."
+    fi
+fi
+
+# Check to see if logging namespace provided exists and components have already been deployed
+log_info "Checking for Elasticsearch pods in the $LOG_NS namespace ..."
+if [[ $(kubectl get pods -n $LOG_NS -l app=v4m-es -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+  log_error "No logging components found in the $LOG_NS namespace."
+  log_error "Logging needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+  exit 1
+else
+  log_debug "Logging deployment found in $LOG_NS namespace.  Continuing."
+fi
+
+# get admin credentials
+export ES_ADMIN_USER=$(kubectl -n $LOG_NS get secret internal-user-admin -o=jsonpath="{.data.username}" |base64 --decode)
+export ES_ADMIN_PASSWD=$(kubectl -n $LOG_NS get secret internal-user-admin -o=jsonpath="{.data.password}" |base64 --decode)
+
+get_sec_api_url
+get_credentials_from_secret admin
+
+# FOR TENANT - Check to see if tenant has been deployed in logging.
+if [ "$cluster" != "true" ]; then
+    log_info "Checking that [$tenant] has been onboarded to Elasticsearch in the $LOG_NS namespace ..."
+
+    if ! kibana_tenant_exists "${tenantNS}_${tenant}"; then
+        log_error "Unable to configure Elasticsearch data source."
```

Comment:
"datasource" or "data source"  Check the Grafana doc and let's be sure we're consistent.

Reword the two log messages. Perhaps:
`log_error "Unable to configure a Grafana datasource for this tenant because the log monitoring onboarding process has not been completed for the [$tenant] in the [$tenantNS] namespace."`
`log_error "This can be done by running the logging/onboard.sh script"`

Reply:
I think this fell through the cracks and has not been addressed.